### ISSUE TITLE
Can't get code to work

### ISSUE THREAD

**Issue by thomaspreece** [asousa/DepthPrediction#2]
Hi,
I'm trying to get the code to run but it appears that there are missing files. I cannot find any of the caffe model files needed and I don't understand the training code you have used. Could I get the model files and do you possibly have some slightly more detailed information on how to use the code?
Thanks

**Comment by dfreese** [comment]
The caffe files referenced in setup.sh are from the places database.  Check out [the places database webpage here](http://places.csail.mit.edu/) to download the relevant models.  They are too large to store on this repository.

Unfortunately we do not have additional documentation of the training code, as this project was done on a short timeframe, and isn't actively maintained.

**Comment by asousa** [comment]
Hi Thomas,

I wish I understood it either! That project was very rushed, and we struggled a lot with getting Caffe to cooperate. By the time we turned it in we were _almost_ getting decent results.

I’m pretty sure we were using the Places205 CNN pre-trained model (http://places.csail.mit.edu/downloadCNN.html).

It’s a little obnoxious, but the gist is that we called the caffe train methods from the command line and not from Python.

Training parameters are specified in whichever solver file you’re using (models/solver_softmax_dir.prototxt).

So to train, you run something like:
caffe train -solver <solver file> -weights <pre-trained file> -gpu 0
Then to classify, you do something like the following in Python:
DP = DepthPredictor()
DP.load(model_file, pretrained_file, mean file)
predicted_depth = DP.predict( input image )

The actual model you’re using is in models/udr_.prototxt… I did a few versions - euclidean or softmax loss, and with a few different inputs (either from an LMDB file or a directory of images). To run, there’s a different model file with the input block swapped out (_deploy.prototxt).

Honestly, when we did this Caffe was a lot rougher than it is now - we wasted a lot of time when the syntax for the model files changed halfway through.

On a conceptual note, we never did come to terms with using the neural net to do regression on a continuous variable (depth) vs a classification of quantized values (is it in bin 1 or not, is it in bin 2 or not…).

If you haven’t read it already, check out the paper we were working from - http://arxiv.org/abs/1411.6387 - they got solid results but were using matconvnet. Our whole plan was to duplicate with caffe, and then tweak / train the last few layers.

Good luck, hope this helps!
Austin

On Jun 7, 2016, at 4:54 AM, Thomas Preece <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi,
I'm trying to get the code to run but it appears that there are missing files. I cannot find any of the caffe model files needed and I don't understand the training code you have used. Could I get the model files and do you possibly have some slightly more detailed information on how to use the code?
Thanks

**Comment by iij0** [comment]
Hi
I am also trying to run this code but I don't know how to prepare  for the training and testing data. Could you provide these related files such as "DepthPrediction/train/NYUv2/splits.mat", "train/NYUv2/train_full_167_v3", "inp_file = 'train/NYUv2/test_full_167_v3/index_32_1.txt"
"out_file = 'train/NYUv2/test_full_167_v3/index_32_1_norm_randomized.txt".

Thanks so much

**Comment by luohongcheng** [comment]
Hi, Asousa. I am running your codes this days, and in this codes I hava not find the loss function of this paper. The output of this networks is label only. so I guess if it exists any difference from the paper's method ?

**Comment by chihhsuanwu** [comment]
HI Asousa
The download website for Places205 CNN pre-trained model can not connect
please fix it 
Thanks
